# Dallas Bueckert

## Research question/interests
How does a game's genre affect the rating people give it on the steam store.

In [5]:
import pandas as pd
df = pd.read_csv('../data/raw/steam.csv')
df

appid                        name release_date  english  \
0           10              Counter-Strike   2000-11-01        1   
1           20       Team Fortress Classic   1999-04-01        1   
2           30               Day of Defeat   2003-05-01        1   
3           40          Deathmatch Classic   2001-06-01        1   
4           50   Half-Life: Opposing Force   1999-11-01        1   
...        ...                         ...          ...      ...   
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher          platforms  \
0                   Valve                 Valve  windows;mac;linux   
1                   Valve                 Valve  windows;mac;linux   
2                   Valve                 Valve  windows;mac;linux   
3                   Valve                 Valve  windows;mac;linux   
4        Gearbox Software                 Valve  windows;mac;linux   
...                   ...                   ...                ...   
27070         SHEN JIAWEI           SHEN JIAWEI            windows   
27071      Semyon Maximov      BekkerDev Studio            windows   
27072         EntwicklerX           EntwicklerX            windows   
27073  Yustas Game Studio  Alawar Entertainment        windows;mac   
27074    Adept Studios GD  Alawar Entertainment        windows;mac   

       required_age                                         categories  \
0                 0  Multi-player;Online Multi-Player;Local Multi-P...   
1                 0  Multi-player;Online Multi-Player;Local Multi-P...   
2                 0              Multi-player;Valve Anti-Cheat enabled   
3                 0  Multi-player;Online Multi-Player;Local Multi-P...   
4                 0  Single-player;Multi-player;Valve Anti-Cheat en...   
...             ...                                                ...   
27070             0                   Single-player;Steam Achievements   
27071             0                                      Single-player   
27072             0  Single-player;Multi-player;Co-op;Shared/Split ...   
27073             0                          Single-player;Steam Cloud   
27074             0                          Single-player;Steam Cloud   

                       genres                 steamspy_tags  achievements  \
0                      Action        Action;FPS;Multiplayer             0   
1                      Action        Action;FPS;Multiplayer             0   
2                      Action  FPS;World War II;Multiplayer             0   
3                      Action        Action;FPS;Multiplayer             0   
4                      Action             FPS;Action;Sci-fi             0   
...                       ...                           ...           ...   
27070  Adventure;Casual;Indie        Adventure;Indie;Casual             7   
27071  Action;Adventure;Indie        Action;Indie;Adventure             0   
27072     Action;Casual;Indie           Action;Indie;Casual            24   
27073  Adventure;Casual;Indie        Indie;Casual;Adventure             0   
27074  Adventure;Casual;Indie        Indie;Casual;Adventure             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
0                124534              3339             17612              317   
1                  3318               633               277               62   
2                  3416               398               187               34   
3                  1273               267               258              184   
4                  5250               288               624              415   
...                 ...               ...               .

# Milestone 3:

## Task 1: EDA

In [8]:
import matplotlib.pylab as plt
import seaborn as sn
import numpy as np

print("The number of columns and rows:")
print(df.shape)
print("All of the categories:")
print(df.columns)

The number of columns and rows:
(27075, 18)
All of the categories:
Index(['appid', 'name', 'release_date', 'english', 'developer', 'publisher',
       'platforms', 'required_age', 'categories', 'genres', 'steamspy_tags',
       'achievements', 'positive_ratings', 'negative_ratings',
       'average_playtime', 'median_playtime', 'owners', 'price'],
      dtype='object')


In [20]:
df.describe(include='int')

appid       english  required_age  achievements  \
count  2.707500e+04  27075.000000  27075.000000  27075.000000   
mean   5.962035e+05      0.981127      0.354903     45.248864   
std    2.508942e+05      0.136081      2.406044    352.670281   
min    1.000000e+01      0.000000      0.000000      0.000000   
25%    4.012300e+05      1.000000      0.000000      0.000000   
50%    5.990700e+05      1.000000      0.000000      7.000000   
75%    7.987600e+05      1.000000      0.000000     23.000000   
max    1.069460e+06      1.000000     18.000000   9821.000000   

       positive_ratings  negative_ratings  average_playtime  median_playtime  
count      2.707500e+04      27075.000000      27075.000000      27075.00000  
mean       1.000559e+03        211.027147        149.804949        146.05603  
std        1.898872e+04       4284.938531       1827.038141       2353.88008  
min        0.000000e+00          0.000000          0.000000          0.00000  
25%        6.000000e+00          2.000000          0.000000          0.00000  
50%        2.400000e+01          9.000000          0.000000          0.00000  
75%        1.260000e+02         42.000000          0.000000          0.00000  
max        2.644404e+06     487076.000000     190625.000000     190625.00000

###### 98.1% of games uploaded to steam during this time had english as a language.
###### 25% of games only have 6 positive ratings or less, half of all games have 24 positive ratingas or less, and 75% of all games only have 126 positive ratings or less. Very few games are very or even moderatley succesfull.
###### 75% of games have an average of 0 hrs played while the mean is around 150hrs, showing that some games are played significantly more than others. Even though there is a skew of the mean given the max average playtime is 190625 the mean median is still close at 146hrs.

In [21]:
df.describe(include='float64').T

count      mean       std  min   25%   50%   75%     max
price  27075.0  6.078193  7.874922  0.0  1.69  3.99  7.19  421.99

###### 75% of the games cost $7.19 USD or less, I wonder if this correlates with positive ratings?

In [19]:
df.describe(include='object')

name release_date        developer       publisher platforms  \
count         27075        27075            27075           27075     27075   
unique        27033         2619            17113           14354         7   
top     Dark Matter   2018-07-13  Choice of Games  Big Fish Games   windows   
freq              3           64               94             212     18398   

           categories        genres        steamspy_tags   owners  
count           27075         27075                27075    27075  
unique           3333          1552                 6423       13  
top     Single-player  Action;Indie  Action;Indie;Casual  0-20000  
freq             6110          1852                  845    18596

###### There are 1552 unique genres here but I will probably only use the top 10 for my research question. I can also probably use the top 10 categories as they are similar enough to genres.

## Task 2-3: Set up an Analysis Pipeline by using Method Chaining and writing Python programs

### Load in unproccessed data:

### Load in unprocessed data and clean, process, and wrangle it:

## Task 4: Conduct Analysis and Answer my Research Question: